### PR TITLE
Update README.so.md to sync with the English one

### PR DESCRIPTION
**Pull request by nadiration**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [x] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

Synced with the English Readme + some minor improvements

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: README.so.md [hunk 1]
```diff
@@ -12,13 +12,13 @@
 <a href="https://www.bountysource.com/teams/newpipe" alt="Kuwa Bountysource "><img src="https://img.shields.io/bountysource/team/newpipe/activity.svg?colorB=cd201f"></a>
 </p>
 <hr>
-<p align="center"><a href="#screenshots">Sawir-shaashadeed</a> &bull; <a href="#description">Faahfaahin</a> &bull; <a href="#features">Waxqabadka</a> &bull; <a href="#updates">Cusboonaysiin</a> &bull; <a href="#contribution">Kusoo Kordhin</a> &bull; <a href="#donate">Ugu Deeq</a> &bull; <a href="#license">Laysinka</a></p>
+<p align="center"><a href="#screenshots">Sawir-shaashadeed</a> &bull; <a href="#description">Faahfaahin</a> &bull; <a href="#features">Waxqabadka</a> &bull; <a href="#updates">Kushubida iyo cusboonaysiinta</a> &bull; <a href="#contribution">Kusoo Kordhin</a> &bull; <a href="#donate">Ugu Deeq</a> &bull; <a href="#license">Laysinka</a></p>
```
Comment (Stypox) [inline README.so.md:15]: The word after the `#` (e.g. `screenshots` in `<a href="#screenshots">`) should point to the correct section (e.g `<a href="#sawir-shaashadeed">`)

**Comment by nadiration** [comment]
@Stypox implemented requested changes. could you review it?

**Review by Stypox — APPROVED** [review]
@mhmdanas I think this can be merged. Thank you @nadiration :-)

**Comment by opusforlife2** [comment]
Thanks, @nadiration!

**Pull request merged**